pipelines/py/build-installer.yaml

```diff
@@ -0,0 +1,52 @@
+name: Build and install python packages with build and installer
+
+inputs:
+  python:
+    description: which python to use
+    default: DEFAULT
+  dest:
+    description: the destination
+    default: ${{targets.contextdir}}
+
+pipeline:
+  - name: "Build installer"
+    runs: |
+      py=${{inputs.python}}
+      if [ "$py" = "DEFAULT" ]; then
+        if p=$(command -v python3); then
+          py=$p
+          if [ -L "$p" ]; then
+            py=$(readlink -f "$p") ||
+              { echo "ERROR: failed 'readlink -f $p'" 1>&2; exit 1; }
+          fi
+        else
+          glob="/usr/bin/python3.[0-9][0-9] /usr/bin/python3.[789]"
+          n=0
+          for p in $glob; do
+            [ -x "$p" ] && n=$((n+1)) && py=$p && found="$found $p"
+          done
+          if [ "$n" -ne 1 ]; then
+            echo "ERROR: must set inputs.python: " \
+                "found $n executables matching $glob" 1>&2
+            [ "$n" -eq 0 ] || echo "  found: $found" 1>&2
+            exit 1
+          fi
+        fi
+      fi
+      pyver=$("$py" -c 'import sys; print("%s.%s" % (sys.version_info.major, sys.version_info.minor))')
+      sitepkgd=$("$py" -c 'import site; print(site.getsitepackages()[0])')
+      sitepkgd=${sitepkgd#/}
+      root=${{inputs.dest}}
+      wd="./.wheels/$pyver"
+      vr() { echo "execute:" "$@"; "$@"; }
+
+      tmpd=$(mktemp -d)
+      trap "rm -Rf '$tmpd'" EXIT
+
+      # --find-links to an empty dir and --no-index makes pip fully "offline"
+      distwheelsd="$tmpd/dist-wheels"
+      mkdir -p "$distwheelsd"
+      echo "$py is $pyver with site_packages dir '$sitepkgd'"
+      vr $py -m build --wheel --verbose --no-isolation
+      vr $py -m installer -d $root dist/*.whl
+      vr find $root -name "*.pyc" -exec rm -rf '{}' +
```

Comment:
this is wrong. we do not want to remove pyc files.

Reply:
Ahh OK I can remove that, It does seem to be fairly common in our files though